### PR TITLE
feat(ci): setup full pipeline

### PR DESCRIPTION
**Pull request by MickVanDuijn**

### CODE REVIEW

File: .github/workflows/typescript.yml [hunk 1]
```diff
@@ -2,53 +2,71 @@ name: Node CI
...
-    #     yarn npm-defaults release
-    #   env:
-    #     GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
-    #     NPM_TOKEN: ${{secrets.NPM_TOKEN}}
-    # - name: create pull request
-    #   if: github.ref == 'refs/heads/development'
-    #   run: yarn npm-defaults make-release
-    #   env:
-    #     GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (pevisscher) [inline .github/workflows/typescript.yml:71]: do we still need this?

File: .npmrc [hunk 2]
```diff
@@ -1,3 +1,4 @@
-@zefiros-software:registry=https://npm.pkg.github.com
+//npm.pkg.github.com/:_authToken=${GITHUB_NPM_TOKEN}
```
Comment (pevisscher) [inline .npmrc:1]: why not `GITHUB_TOKEN`?

Comment (MickVanDuijn) [reply .npmrc:1]: this was the token I was talking about where we needed to align on naming. my original reasoning: so we don't mix it up on local development environments with personal access tokens that we use on the Github CLI. we can keep this `GITHUB_NPM_TOKEN` locally as a read-only token that has limited scopes. but I'm happy to consider alternatives 😁

File: .github/workflows/typescript.yml (hunk 1 above)
Comment (MickVanDuijn) [reply .github/workflows/typescript.yml:71]: gooood question, I'll remove it and see if the one on the top-level of the pipeline is sufficient 👍

Comment (MickVanDuijn) [reply .github/workflows/typescript.yml]: yeap, we need it for the releases, added a comment 👌

File: .npmrc (hunk 2 above)
Comment (MickVanDuijn) [reply .npmrc:1]: as discussed offline, this is initially what we'll go for!

**Pull request merged**